pkg/kubelet/config/etcd.go

```diff
@@ -59,7 +59,7 @@ func NewSourceEtcd(key string, client tools.EtcdClient, updates chan<- interface
 }
 
 func (s *sourceEtcd) run() {
-	watching := s.helper.Watch(s.key, 0)
+	watching := s.helper.Watch(s.key, 0, true /* emitNilOnNotFound */)
```

Comment:
So I know we chatted about this last night, but I think I wasn't thinking very clearly. I think the pattern we decided on elsewhere, basically because of this problem, is that clients must first list, and then watch (using the resourceVersion the list returns).

So here is an example (though it uses the api and not etcd): https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/proxy/config/api.go#L82

s.helper should have a few different List methods that will return what you need.

Sorry for the bad advice last night, I didn't make the right connection until I saw this written out :/